server/app.js

```diff
@@ -0,0 +1,19 @@
+const express = require('express');
+const app = express();
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+app.use(express.static(__dirname + '/../public'));
+// app.use(express.static(__dirname + '/../client/dist'));
+
+app.get('/', (req, res, err) => {
+    res.send(data);
+});
+
+
+let port = process.env.PORT || 3017;
+app.listen(port, function () {
+    console.log(`Listening on port ${port}`);    
+});
+
+module.exports = app;
```

Comment:
A single page application will need to send back the proxy index.html file every time the path in the URL is changed.  
The way this proxy is set up will never handle any of the React Components JavaScript. This is a "dumb" proxy which should only renders the main `index.html` file.  Create a path for any route that you expect to be handled, which in turn returns that file.

If you'd like to use a Route in express to handle any direction to `/:id` there are two ways to achieve this.

1. create an `app.get` route to handle this path.  The route must return the index.html file
2. create a "Catch All" route which effectively will catch any path not specifically declared. The route must return the index.html file.  You can use the asterisk as a catch all. Example: `app.get('*', callback)`